attribution/api_attribution.py

```diff
@@ -112,32 +107,53 @@ def compute_attributions(self, input_text: str, **kwargs):
                 for unit_token_ids in token_ids_per_unit[i_unit + 1 :]
                 for token_id in unit_token_ids
             ]
-            perturbed_input = self.local_tokenizer.decode(
-                left_token_ids + replacement_token_ids + right_token_ids
+            perturbed_input = self.tokenizer.decode(
+                left_token_ids + replacement_token_ids + right_token_ids, skip_special_tokens=True
             )
 
             # Get the output logprobs for the perturbed input
             perturbed_output = self.get_chat_completion(perturbed_input)
 
+
+            if ignore_output_token_location:
+
+                all_top_logprobs = []
+                all_toks = []
+                for ptl in perturbed_output.logprobs.content:
+                    all_top_logprobs.extend([tl.logprob for tl in ptl.top_logprobs])
+                    all_toks.extend([tl.token for tl in ptl.top_logprobs])
+
+                sorted_indexes = sorted(range(len(all_top_logprobs)), key=all_top_logprobs.__getitem__, reverse=True)
+                all_toks = [all_toks[s] for s in sorted_indexes]
+                all_top_logprobs = [all_top_logprobs[s] for s in sorted_indexes]
+
+                for otl in remaining_output.logprobs.content:
```

Comment:
If this is for the remaining_output top logprobs, maybe `ro_top_logprobs` or something that indicates what the `o` stands for, as there are multiple output variables in this context (original, perturbed, remaining).